modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -144,46 +82,79 @@ function AddFDSFilterModalContent({
 	namespace,
 	onSave,
 }: IPropsAddFDSFilterModalContent) {
-	const [selectedField, setSelectedField] = useState<string>();
-	const [label, setLabel] = useState<string>();
-	const [from, setFrom] = useState<string>('');
-	const [to, setTo] = useState<string>('');
+	const fromFormElementId = `${namespace}From`;
+	const toFormElementId = `${namespace}To`;
+	const nameFormElementId = `${namespace}Name`;
+	const selectedFieldFormElementId = `${namespace}SelectedField`;
+	const sourceOptionFormElementId = `${namespace}SourceOption`;
+	const multipleFormElementId = `${namespace}Multiple`;
+	const includeModeFormElementId = `${namespace}IncludeMode`;
+	const preselectedValuesFormElementId = `${namespace}PreselectedValues`;
+
+	const [picklists, setPicklists] = useState<IPickList[]>([]);
+	const [multiSelectValues, setMultiSelectValues] = useState<any[]>([]);
+
+	useEffect(() => {
+		getAllPicklists().then((items) => setPicklists(items));
+	}, []);
+
+	const [formValues, setFormValues] = useState({
+		[fromFormElementId]: '',
+		[includeModeFormElementId]: 'true',
+		[multipleFormElementId]: 'true',
```

Comment:
I'm not happy about this. It is because radio's can only have a value of type `string`. I want to refactor our forms with something like `formik` to make them more straight forward, but I had some major TS headaches with that. I want to look more into  form management next week though